models/issue_tracked_time.go

```diff
@@ -131,3 +186,114 @@ func TotalTimes(options FindTrackedTimesOptions) (map[*User]string, error) {
 	}
 	return totalTimes, nil
 }
+
+// DeleteIssueUserTimes deletes times for issue
+func DeleteIssueUserTimes(issue *Issue, user *User) error {
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	opts := FindTrackedTimesOptions{
+		IssueID: issue.ID,
+		UserID:  user.ID,
+	}
+
+	removedTime, err := GetTrackedSeconds(opts)
+	if err != nil {
+		return err
+	}
+	if removedTime == 0 {
+		return ErrNotExist{}
+	}
+
+	removedTime, err = deleteTimes(sess, opts)
+	if err != nil {
+		return err
+	}
+
+	if err := issue.loadRepo(sess); err != nil {
+		return err
+	}
+	if _, err := createComment(sess, &CreateCommentOptions{
+		Issue:   issue,
+		Repo:    issue.Repo,
+		Doer:    user,
+		Content: "- " + SecToTime(removedTime),
+		Type:    CommentTypeDeleteTimeManual,
+	}); err != nil {
+		return err
+	}
+
+	err = sess.Commit()
+	return err
+}
+
+func deleteTimes(e Engine, opts FindTrackedTimesOptions) (removedTime int64, err error) {
+	tt, err := GetTrackedTimes(opts)
+	if err != nil {
+		return
+	}
+
+	for _, t := range tt {
+		_, err = e.Delete(t)
```

Comment:
If you persist in this structure you have to do all the deletes within a single session transaction. This creates a new session each time